app/bundles/LeadBundle/Controller/Api/LeadApiController.php

```diff
@@ -205,6 +205,106 @@ public function getNotesAction($id)
     }
 
     /**
+     * Obtains DonotContact status on a specific Contact
+     *
+     * @param $id
+     *
+     * @return \Symfony\Component\HttpFoundation\Response
+     */
+    public function getDoNotEmailAction($id)
```

Comment:
DoNotContact is now supported by multiple channels (email, sms, etc). So let's not limit this API endpoint to just emails. Rather use the existing LeadModel methods so the same endpoint point can be used across all channels. (Accept a channel and channel_id via parameters). Using `LeadModel::addDncForLead` should suffice as it will check if the contact already has an entry for the given channel, check that the status is different than the what's current set before creating a new entry, and ensures that the API cannot re-subscribe a contact against their wishes if they unsubscribed themselves (all this is already handled by addDncForLead).